exonum-java-binding-cryptocurrency-demo/frontend/src/plugins/blockchain.js

```diff
@@ -0,0 +1,184 @@
+import * as Exonum from 'exonum-client'
+import axios from 'axios'
+import * as Protobuf from 'protobufjs/light'
+import bigInt from 'big-integer'
+import nacl from 'tweetnacl'
+
+var Root  = Protobuf.Root,
+    Type  = Protobuf.Type,
+    Field = Protobuf.Field;
+
+
+let CreateTransactionProtobuf = new Type("CreateTransaction").add(new Field("ownerPublicKey", 1, "bytes"));
+CreateTransactionProtobuf.add(new Field("initialBalance", 2, "int64"))
+
+let TransferTransactionProtobuf = new Type("TransferTransaction").add(new Field("seed", 1, "u-int64"));
+TransferTransactionProtobuf.add(new Field("senderId", 2, "bytes"))
+TransferTransactionProtobuf.add(new Field("recipientId", 3, "bytes"))
+TransferTransactionProtobuf.add(new Field("amount", 4, "int64"))
+
+var root = new Root();
+root.define("CreateTransactionProtobuf").add(CreateTransactionProtobuf);
+root.define("TransferTransactionProtobuf").add(TransferTransactionProtobuf);
+
+const TX_URL = '/api/cryptocurrency-demo-service/submit-transaction'// '/api/services/cryptocurrency/v1/wallets/transaction'
+const PER_PAGE = 10
+const ATTEMPTS = 10
+const ATTEMPT_TIMEOUT = 500
+const PROTOCOL_VERSION = 0
+const SERVICE_ID = 42
+const TX_TRANSFER_ID = 2
+const TX_WALLET_ID = 1
+
+const MessageHead = Exonum.newType({
+  fields: [
+    { name: 'network_id', type: Exonum.Uint8 },
+    { name: 'protocol_version', type: Exonum.Uint8 },
+    { name: 'message_id', type: Exonum.Uint16 },
+    { name: 'service_id', type: Exonum.Uint16 },
+    { name: 'payload', type: Exonum.Uint32 }
+  ]
+})
+
+function getWallet(publicKey) {
+  return axios.get('/api/services/configuration/v1/configs/actual').then(response => {
+    // actual list of public keys of validators
+    const validators = response.data.config.validator_keys.map(validator => {
+      return validator.consensus_key
+    })
+    
+    return axios.get(`/api/cryptocurrency-demo-service/wallet/${publicKey}`)
+      .then(response => response.data)
+      .then(data => {
+        const wallet = data
+
+        return {
+          //block: data.block_proof.block,
+          wallet: wallet,
+          transactions: []
+        } 
+      })
+  })
+}
+
+function waitForAcceptance(publicKey, hash) {
+  let attempt = ATTEMPTS
+
+  return (function makeAttempt() {
+    return axios.get(`/api/explorer/v1/transactions/${hash}`).then(response =>  {
+      if (response.data.type !== 'committed') {
+        if (--attempt > 0) {
+          return new Promise(resolve => {
+            setTimeout(resolve, ATTEMPT_TIMEOUT)
+          }).then(makeAttempt)
+        } else {
+          throw new Error('Transaction has not been found')
+        }
+      } else {
+        return response.data
+      }
+    })
+  })()
+}
+
+module.exports = {
+  install(Vue) {
+    Vue.prototype.$blockchain = {
+      generateKeyPair() {
+        return Exonum.keyPair()
+      },
+
+      generateSeed() {
+        const buffer = nacl.randomBytes(7)
```

Comment:
What is 7 — a seed, size, anything else? I'd suggest giving this constant a name.

Reply:
@dmitry-timofeev honestly it should be random integer of `8` bytes here, which is 2^64, but as soon as Java supports only 2^63 integers we decided to do this hack..